src/cpp/include/openvino/genai/vision_encoder.hpp

```diff
@@ -41,11 +42,16 @@ struct EncodedImage {
 /// ov::InferRequest and configured by ProcessorConfig.
 class OPENVINO_GENAI_EXPORTS VisionEncoder {
 public:
+    /// @brief A enum denoting model type.
+    VLMModelType model_type;
     /// @brief A model for image encoding.
     ov::InferRequest m_encoder;
     /// @brief A config to follow.
     ProcessorConfig m_processor_config;
 
+    // LLaVa specific members
+    ov::InferRequest m_vision_embeddings;
```

Comment:
Why isn't `m_encoder` reused? You can rename it if you feel `m_vision_embeddings` is a better name.

Reply:
Fixed